# Climate Change

There have been many studies documenting that the average global temperature has been increasing over the last century. The consequences of a continued rise in global temperature will be dire. Rising sea levels and an increased frequency of extreme weather events will affect billions of people.

In this problem, we will attempt to study the relationship between average global temperature and several other factors.

The file contains climate data from May 1983 to December 2008.

<img src="images/0219.jpg"/>

Year: the observation year.

Month: the observation month.

Temp: the difference in degrees Celsius between the average global temperature in that period and a reference value. This data comes from the Climatic Research Unit at the University of East Anglia.

CO2, N2O, CH4, CFC.11, CFC.12: atmospheric concentrations of carbon dioxide (CO2), nitrous oxide (N2O), methane  (CH4), trichlorofluoromethane (CCl3F; commonly referred to as CFC-11) and dichlorodifluoromethane (CCl2F2; commonly referred to as CFC-12), respectively. This data comes from the ESRL/NOAA Global Monitoring Division.

    CO2, N2O and CH4 are expressed in ppmv (parts per million by volume  -- i.e., 397 ppmv of CO2 means that CO2 constitutes 397 millionths of the total volume of the atmosphere)
    
    CFC.11 and CFC.12 are expressed in ppbv (parts per billion by volume). 

Aerosols: the mean stratospheric aerosol optical depth at 550 nm. This variable is linked to volcanoes, as volcanic eruptions result in new particles being added to the atmosphere, which affect how much of the sun's energy is reflected back into space. This data is from the Godard Institute for Space Studies at NASA.

TSI: the total solar irradiance (TSI) in W/m2 (the rate at which the sun's energy is deposited per unit area). Due to sunspots and other solar phenomena, the amount of energy that is given off by the sun varies substantially with time. This data is from the SOLARIS-HEPPA project website.

MEI: multivariate El Nino Southern Oscillation index (MEI), a measure of the strength of the El Nino/La Nina-Southern Oscillation (a weather effect in the Pacific Ocean that affects global temperatures). This data comes from the ESRL/NOAA Physical Sciences Division.

### Read in data

In [1]:
CC = read.csv("data/climate_change.csv")
head(CC)

Year Month MEI   CO2    CH4     N2O     CFC.11  CFC.12  TSI      Aerosols
1 1983  5    2.556 345.96 1638.59 303.677 191.324 350.113 1366.102 0.0863  
2 1983  6    2.167 345.52 1633.71 303.746 192.057 351.848 1366.121 0.0794  
3 1983  7    1.741 344.15 1633.22 303.795 192.818 353.725 1366.285 0.0731  
4 1983  8    1.130 342.25 1631.35 303.839 193.602 355.633 1366.420 0.0673  
5 1983  9    0.428 340.17 1648.40 303.901 194.392 357.465 1366.234 0.0619  
6 1983 10    0.002 340.30 1663.79 303.970 195.171 359.174 1366.059 0.0569  
  Temp 
1 0.109
2 0.118
3 0.137
4 0.176
5 0.149
6 0.093

In [2]:
str(CC)

'data.frame':	308 obs. of  11 variables:
 $ Year    : int  1983 1983 1983 1983 1983 1983 1983 1983 1984 1984 ...
 $ Month   : int  5 6 7 8 9 10 11 12 1 2 ...
 $ MEI     : num  2.556 2.167 1.741 1.13 0.428 ...
 $ CO2     : num  346 346 344 342 340 ...
 $ CH4     : num  1639 1634 1633 1631 1648 ...
 $ N2O     : num  304 304 304 304 304 ...
 $ CFC.11  : num  191 192 193 194 194 ...
 $ CFC.12  : num  350 352 354 356 357 ...
 $ TSI     : num  1366 1366 1366 1366 1366 ...
 $ Aerosols: num  0.0863 0.0794 0.0731 0.0673 0.0619 0.0569 0.0524 0.0486 0.0451 0.0416 ...
 $ Temp    : num  0.109 0.118 0.137 0.176 0.149 0.093 0.232 0.078 0.089 0.013 ...


In [3]:
summary(CC)

      Year          Month             MEI               CO2       
 Min.   :1983   Min.   : 1.000   Min.   :-1.6350   Min.   :340.2  
 1st Qu.:1989   1st Qu.: 4.000   1st Qu.:-0.3987   1st Qu.:353.0  
 Median :1996   Median : 7.000   Median : 0.2375   Median :361.7  
 Mean   :1996   Mean   : 6.552   Mean   : 0.2756   Mean   :363.2  
 3rd Qu.:2002   3rd Qu.:10.000   3rd Qu.: 0.8305   3rd Qu.:373.5  
 Max.   :2008   Max.   :12.000   Max.   : 3.0010   Max.   :388.5  
      CH4            N2O            CFC.11          CFC.12           TSI      
 Min.   :1630   Min.   :303.7   Min.   :191.3   Min.   :350.1   Min.   :1365  
 1st Qu.:1722   1st Qu.:308.1   1st Qu.:246.3   1st Qu.:472.4   1st Qu.:1366  
 Median :1764   Median :311.5   Median :258.3   Median :528.4   Median :1366  
 Mean   :1750   Mean   :312.4   Mean   :252.0   Mean   :497.5   Mean   :1366  
 3rd Qu.:1787   3rd Qu.:317.0   3rd Qu.:267.0   3rd Qu.:540.5   3rd Qu.:1366  
 Max.   :1814   Max.   :322.2   Max.   :271.5   Max.   :5

### Problem 1.1 - Creating Our First Model

We are interested in how changes in these variables affect future temperatures, as well as how well these variables explain temperature changes so far. To do this, first read the dataset climate_change.csv into R.

Then, split the data into a training set, consisting of all the observations up to and including 2006, and a testing set consisting of the remaining years (bhint: use subset). A training set refers to the data that will be used to build the model (this is the data we give to the lm() function), and a testing set refers to the data we will use to test our predictive ability.

Next, build a linear regression model to predict the dependent variable Temp, using MEI, CO2, CH4, N2O, CFC.11, CFC.12, TSI, and Aerosols as independent variables (Year and Month should NOT be used in the model). Use the training set to build the model.

Enter the model R2 (the "Multiple R-squared" value)

In [4]:
Train <- subset(CC, CC$Year<=2006)
Test <- subset(CC, CC$Year>2006)

In [5]:
head(Train)

Year Month MEI   CO2    CH4     N2O     CFC.11  CFC.12  TSI      Aerosols
1 1983  5    2.556 345.96 1638.59 303.677 191.324 350.113 1366.102 0.0863  
2 1983  6    2.167 345.52 1633.71 303.746 192.057 351.848 1366.121 0.0794  
3 1983  7    1.741 344.15 1633.22 303.795 192.818 353.725 1366.285 0.0731  
4 1983  8    1.130 342.25 1631.35 303.839 193.602 355.633 1366.420 0.0673  
5 1983  9    0.428 340.17 1648.40 303.901 194.392 357.465 1366.234 0.0619  
6 1983 10    0.002 340.30 1663.79 303.970 195.171 359.174 1366.059 0.0569  
  Temp 
1 0.109
2 0.118
3 0.137
4 0.176
5 0.149
6 0.093

In [6]:
head(Test)

Year Month MEI    CO2    CH4     N2O     CFC.11  CFC.12  TSI      Aerosols
285 2007 1      0.974 382.93 1799.66 320.561 248.372 539.206 1365.717 0.0054  
286 2007 2      0.510 383.81 1803.08 320.571 248.264 538.973 1365.715 0.0051  
287 2007 3      0.074 384.56 1803.10 320.548 247.997 538.811 1365.754 0.0045  
288 2007 4     -0.049 386.40 1802.11 320.518 247.574 538.586 1365.723 0.0045  
289 2007 5      0.183 386.58 1795.65 320.445 247.224 538.130 1365.693 0.0041  
290 2007 6     -0.358 386.05 1781.81 320.332 246.881 537.376 1365.762 0.0040  
    Temp 
285 0.601
286 0.498
287 0.435
288 0.466
289 0.372
290 0.382

In [7]:
str(Train)

'data.frame':	284 obs. of  11 variables:
 $ Year    : int  1983 1983 1983 1983 1983 1983 1983 1983 1984 1984 ...
 $ Month   : int  5 6 7 8 9 10 11 12 1 2 ...
 $ MEI     : num  2.556 2.167 1.741 1.13 0.428 ...
 $ CO2     : num  346 346 344 342 340 ...
 $ CH4     : num  1639 1634 1633 1631 1648 ...
 $ N2O     : num  304 304 304 304 304 ...
 $ CFC.11  : num  191 192 193 194 194 ...
 $ CFC.12  : num  350 352 354 356 357 ...
 $ TSI     : num  1366 1366 1366 1366 1366 ...
 $ Aerosols: num  0.0863 0.0794 0.0731 0.0673 0.0619 0.0569 0.0524 0.0486 0.0451 0.0416 ...
 $ Temp    : num  0.109 0.118 0.137 0.176 0.149 0.093 0.232 0.078 0.089 0.013 ...


In [8]:
str(Test)

'data.frame':	24 obs. of  11 variables:
 $ Year    : int  2007 2007 2007 2007 2007 2007 2007 2007 2007 2007 ...
 $ Month   : int  1 2 3 4 5 6 7 8 9 10 ...
 $ MEI     : num  0.974 0.51 0.074 -0.049 0.183 ...
 $ CO2     : num  383 384 385 386 387 ...
 $ CH4     : num  1800 1803 1803 1802 1796 ...
 $ N2O     : num  321 321 321 321 320 ...
 $ CFC.11  : num  248 248 248 248 247 ...
 $ CFC.12  : num  539 539 539 539 538 ...
 $ TSI     : num  1366 1366 1366 1366 1366 ...
 $ Aerosols: num  0.0054 0.0051 0.0045 0.0045 0.0041 0.004 0.004 0.0041 0.0042 0.0041 ...
 $ Temp    : num  0.601 0.498 0.435 0.466 0.372 0.382 0.394 0.358 0.402 0.362 ...


### Model for Climate Change

In [9]:
Mod1 = lm(Temp ~ MEI + CO2 + CH4 + N2O + CFC.11 + CFC.12 + TSI + Aerosols, data=Train)
summary(Mod1)


Call:
lm(formula = Temp ~ MEI + CO2 + CH4 + N2O + CFC.11 + CFC.12 + 
    TSI + Aerosols, data = Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.25888 -0.05913 -0.00082  0.05649  0.32433 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) -1.246e+02  1.989e+01  -6.265 1.43e-09 ***
MEI          6.421e-02  6.470e-03   9.923  < 2e-16 ***
CO2          6.457e-03  2.285e-03   2.826  0.00505 ** 
CH4          1.240e-04  5.158e-04   0.240  0.81015    
N2O         -1.653e-02  8.565e-03  -1.930  0.05467 .  
CFC.11      -6.631e-03  1.626e-03  -4.078 5.96e-05 ***
CFC.12       3.808e-03  1.014e-03   3.757  0.00021 ***
TSI          9.314e-02  1.475e-02   6.313 1.10e-09 ***
Aerosols    -1.538e+00  2.133e-01  -7.210 5.41e-12 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.09171 on 275 degrees of freedom
Multiple R-squared:  0.7509,	Adjusted R-squared:  0.7436 
F-statistic: 103.6 on 8 and 275 DF,  p-val

#### R-squared

In [10]:
summary(Mod1)$r.square

[1] 0.7508933

### Problem 1.2 - Creating Our First Model

**Which variables are significant in the model?** We will consider a variable signficant only if the p-value is below 0.05. (Select all that apply.)

In [11]:
coef(summary(Mod1))[, "Pr(>|t|)"]

(Intercept)          MEI          CO2          CH4          N2O       CFC.11 
1.431046e-09 4.898887e-20 5.052521e-03 8.101456e-01 5.466931e-02 5.957288e-05 
      CFC.12          TSI     Aerosols 
2.097199e-04 1.095945e-09 5.411273e-12

MEI, CO2, CFC.11, CFC.12, TSI, and Aerosols are all significant.

### Problem 2.1 - Understanding the Model

Current scientific opinion is that nitrous oxide and CFC-11 are greenhouse gases: gases that are able to trap heat from the sun and contribute to the heating of the Earth. However, the regression coefficients of both the N2O and CFC-11 variables are negative, indicating that increasing atmospheric concentrations of either of these two compounds is associated with lower global temperatures.

**Which of the following is the simplest correct explanation for this contradiction?**

All of the gas concentration variables reflect human development - N2O and CFC.11 are correlated with other variables in the data set.

**Compute the correlations between all the variables in the training set. Which of the following independent variables is N2O highly correlated with (absolute correlation greater than 0.7)? Select all that apply.**

In [12]:
cor(Train)

Year        Month         MEI           CO2         CH4        
Year      1.00000000 -0.0279419602 -0.0369876842  0.98274939  0.91565945
Month    -0.02794196  1.0000000000  0.0008846905 -0.10673246  0.01856866
MEI      -0.03698768  0.0008846905  1.0000000000 -0.04114717 -0.03341930
CO2       0.98274939 -0.1067324607 -0.0411471651  1.00000000  0.87727963
CH4       0.91565945  0.0185686624 -0.0334193014  0.87727963  1.00000000
N2O       0.99384523  0.0136315303 -0.0508197755  0.97671982  0.89983864
CFC.11    0.56910643 -0.0131112236  0.0690004387  0.51405975  0.77990402
CFC.12    0.89701166  0.0006751102  0.0082855443  0.85268963  0.96361625
TSI       0.17030201 -0.0346061935 -0.1544919227  0.17742893  0.24552844
Aerosols -0.34524670  0.0148895406  0.3402377871 -0.35615480 -0.26780919
Temp      0.78679714 -0.0998567411  0.1724707512  0.78852921  0.70325502
         N2O         CFC.11      CFC.12        TSI         Aerosols   
Year      0.99384523  0.56910643  0.8970116635  0.17030201 -0.34524670
Month     0.01363153 -0.01311122  0.0006751102 -0.03460619  0.01488954
MEI      -0.05081978  0.06900044  0.0082855443 -0.15449192  0.34023779
CO2       0.97671982  0.51405975  0.8526896272  0.17742893 -0.35615480
CH4       0.89983864  0.77990402  0.9636162478  0.24552844 -0.26780919
N2O       1.00000000  0.52247732  0.8679307757  0.19975668 -0.33705457
CFC.11    0.52247732  1.00000000  0.8689851828  0.27204596 -0.04392120
CFC.12    0.86793078  0.86898518  1.0000000000  0.25530281 -0.22513124
TSI       0.19975668  0.27204596  0.2553028138  1.00000000  0.05211651
Aerosols -0.33705457 -0.04392120 -0.2251312440  0.05211651  1.00000000
Temp      0.77863893  0.40771029  0.6875575483  0.24338269 -0.38491375
         Temp       
Year      0.78679714
Month    -0.09985674
MEI       0.17247075
CO2       0.78852921
CH4       0.70325502
N2O       0.77863893
CFC.11    0.40771029
CFC.12    0.68755755
TSI       0.24338269
Aerosols -0.38491375
Temp      1.00000000

In [13]:
abs(cor(Train)) > 0.7

Year  Month MEI   CO2   CH4   N2O   CFC.11 CFC.12 TSI   Aerosols Temp 
Year      TRUE FALSE FALSE  TRUE  TRUE  TRUE FALSE   TRUE  FALSE FALSE     TRUE
Month    FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  FALSE  FALSE FALSE    FALSE
MEI      FALSE FALSE  TRUE FALSE FALSE FALSE FALSE  FALSE  FALSE FALSE    FALSE
CO2       TRUE FALSE FALSE  TRUE  TRUE  TRUE FALSE   TRUE  FALSE FALSE     TRUE
CH4       TRUE FALSE FALSE  TRUE  TRUE  TRUE  TRUE   TRUE  FALSE FALSE     TRUE
N2O       TRUE FALSE FALSE  TRUE  TRUE  TRUE FALSE   TRUE  FALSE FALSE     TRUE
CFC.11   FALSE FALSE FALSE FALSE  TRUE FALSE  TRUE   TRUE  FALSE FALSE    FALSE
CFC.12    TRUE FALSE FALSE  TRUE  TRUE  TRUE  TRUE   TRUE  FALSE FALSE    FALSE
TSI      FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE   TRUE FALSE    FALSE
Aerosols FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE  FALSE  TRUE    FALSE
Temp      TRUE FALSE FALSE  TRUE  TRUE  TRUE FALSE  FALSE  FALSE FALSE     TRUE

(NO2): Year, CO2, CH4, CFC.12, Temp. 

**Which of the following independent variables is CFC.11 highly correlated with?**

(CFC.11): CH4, CFC.12.

### Problem 3 - Simplifying the Model

Given that the correlations are so high, let us focus on the N2O variable and build a model with only MEI, TSI, Aerosols and N2O as independent variables. Remember to use the training set to build the model.

In [15]:
Mod2 = lm(Temp ~  N2O + MEI + TSI + Aerosols, data=Train)
summary(Mod2)


Call:
lm(formula = Temp ~ N2O + MEI + TSI + Aerosols, data = Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.27916 -0.05975 -0.00595  0.05672  0.34195 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) -1.162e+02  2.022e+01  -5.747 2.37e-08 ***
N2O          2.532e-02  1.311e-03  19.307  < 2e-16 ***
MEI          6.419e-02  6.652e-03   9.649  < 2e-16 ***
TSI          7.949e-02  1.487e-02   5.344 1.89e-07 ***
Aerosols    -1.702e+00  2.180e-01  -7.806 1.19e-13 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.09547 on 279 degrees of freedom
Multiple R-squared:  0.7261,	Adjusted R-squared:  0.7222 
F-statistic: 184.9 on 4 and 279 DF,  p-value: < 2.2e-16


**How does this compare to the coefficient in the previous model with all of the variables?**

In [16]:
summary(Mod2)$r.square

[1] 0.7261321

Mod1 = lm(Temp ~ MEI + CO2 + CH4 + N2O + CFC.11 + CFC.12 + TSI + Aerosols, data=Train)

    Adjusted R-squared:  0.7436

Mod2 = lm(Temp ~  N2O + MEI + TSI + Aerosols, data=Train)

    Adjusted R-squared:  0.7222 

### Problem 4 - Automatically Building the Model

We have many variables in this problem, and as we have seen above, dropping some from the model does not decrease model quality. R provides a function, step, that will automate the procedure of trying different combinations of variables to find a good compromise of model simplicity and R2. This trade-off is formalized by the Akaike information criterion (AIC) - it can be informally thought of as the quality of the model with a penalty for the number of variables in the model.

The step function has one argument - the name of the initial model. It returns a simplified model. Use the step function in R to derive a new model, with the full model as the initial model (HINT: If your initial full model was called "climateLM", you could create a new model with the step function by typing step(climateLM). Be sure to save your new model to a variable name so that you can look at the summary. For more information about the step function, type ?step in your R console.)

Enter the R2 value of the model produced by the step function:

In [17]:
Mod3 <- step(Mod1)
summary(Mod3)

Start:  AIC=-1348.16
Temp ~ MEI + CO2 + CH4 + N2O + CFC.11 + CFC.12 + TSI + Aerosols

           Df Sum of Sq    RSS     AIC
- CH4       1   0.00049 2.3135 -1350.1
<none>                  2.3130 -1348.2
- N2O       1   0.03132 2.3443 -1346.3
- CO2       1   0.06719 2.3802 -1342.0
- CFC.12    1   0.11874 2.4318 -1335.9
- CFC.11    1   0.13986 2.4529 -1333.5
- TSI       1   0.33516 2.6482 -1311.7
- Aerosols  1   0.43727 2.7503 -1301.0
- MEI       1   0.82823 3.1412 -1263.2

Step:  AIC=-1350.1
Temp ~ MEI + CO2 + N2O + CFC.11 + CFC.12 + TSI + Aerosols

           Df Sum of Sq    RSS     AIC
<none>                  2.3135 -1350.1
- N2O       1   0.03133 2.3448 -1348.3
- CO2       1   0.06672 2.3802 -1344.0
- CFC.12    1   0.13023 2.4437 -1336.5
- CFC.11    1   0.13938 2.4529 -1335.5
- TSI       1   0.33500 2.6485 -1313.7
- Aerosols  1   0.43987 2.7534 -1302.7
- MEI       1   0.83118 3.1447 -1264.9



Call:
lm(formula = Temp ~ MEI + CO2 + N2O + CFC.11 + CFC.12 + TSI + 
    Aerosols, data = Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.25770 -0.05994 -0.00104  0.05588  0.32203 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) -1.245e+02  1.985e+01  -6.273 1.37e-09 ***
MEI          6.407e-02  6.434e-03   9.958  < 2e-16 ***
CO2          6.402e-03  2.269e-03   2.821 0.005129 ** 
N2O         -1.602e-02  8.287e-03  -1.933 0.054234 .  
CFC.11      -6.609e-03  1.621e-03  -4.078 5.95e-05 ***
CFC.12       3.868e-03  9.812e-04   3.942 0.000103 ***
TSI          9.312e-02  1.473e-02   6.322 1.04e-09 ***
Aerosols    -1.540e+00  2.126e-01  -7.244 4.36e-12 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.09155 on 276 degrees of freedom
Multiple R-squared:  0.7508,	Adjusted R-squared:  0.7445 
F-statistic: 118.8 on 7 and 276 DF,  p-value: < 2.2e-16


In [18]:
summary(Mod3)$r.square

[1] 0.7508409

**Which of the following variable(s) were eliminated from the full model by the step function?**

CH4

It is interesting to note that the step function does not address the collinearity of the variables, except that adding highly correlated variables will not improve the R2 significantly. The consequence of this is that the step function will not necessarily produce a very interpretable model - just a model that has balanced quality and simplicity for a particular weighting of quality and simplicity (AIC).

### Problem 5 - Testing on Unseen Data

We have developed an understanding of how well we can fit a linear regression to the training data, but does the model quality hold when applied to unseen data?

Using the model produced from the step function, calculate temperature predictions for the testing data set, using the predict function.

Enter the testing set R2:

In [19]:
Test$Predict  <- predict(Mod3, newdata = Test)
Test$Predict

[1] 0.4677808 0.4435404 0.4265541 0.4299162 0.4455113 0.4151422 0.4097367
 [8] 0.3839390 0.3255595 0.3274147 0.3231401 0.3316704 0.3522134 0.3313129
[15] 0.3142112 0.3703410 0.4162213 0.4391458 0.4237965 0.3913679 0.3587615
[22] 0.3451991 0.3607087 0.3638076

In [20]:
Test$Predict

[1] 0.4677808 0.4435404 0.4265541 0.4299162 0.4455113 0.4151422 0.4097367
 [8] 0.3839390 0.3255595 0.3274147 0.3231401 0.3316704 0.3522134 0.3313129
[15] 0.3142112 0.3703410 0.4162213 0.4391458 0.4237965 0.3913679 0.3587615
[22] 0.3451991 0.3607087 0.3638076

In [21]:
Test

Year Month MEI    CO2    CH4     N2O     CFC.11  CFC.12  TSI      Aerosols
285 2007  1     0.974 382.93 1799.66 320.561 248.372 539.206 1365.717 0.0054  
286 2007  2     0.510 383.81 1803.08 320.571 248.264 538.973 1365.715 0.0051  
287 2007  3     0.074 384.56 1803.10 320.548 247.997 538.811 1365.754 0.0045  
288 2007  4    -0.049 386.40 1802.11 320.518 247.574 538.586 1365.723 0.0045  
289 2007  5     0.183 386.58 1795.65 320.445 247.224 538.130 1365.693 0.0041  
290 2007  6    -0.358 386.05 1781.81 320.332 246.881 537.376 1365.762 0.0040  
291 2007  7    -0.290 384.49 1771.89 320.349 246.497 537.113 1365.751 0.0040  
292 2007  8    -0.440 382.00 1779.38 320.471 246.307 537.125 1365.757 0.0041  
293 2007  9    -1.162 380.90 1794.21 320.618 246.214 537.281 1365.716 0.0042  
294 2007 10    -1.142 381.14 1802.38 320.855 246.189 537.380 1365.739 0.0041  
295 2007 11    -1.177 382.42 1803.79 321.062 246.178 537.319 1365.668 0.0042  
296 2007 12    -1.168 383.89 1805.58 321.217 246.261 537.052 1365.693 0.0040  
297 2008  1    -1.011 385.44 1809.92 321.328 246.183 536.876 1365.716 0.0038  
298 2008  2    -1.402 385.73 1803.45 321.345 245.898 536.484 1365.737 0.0036  
299 2008  3    -1.635 385.97 1792.84 321.295 245.430 535.979 1365.673 0.0034  
300 2008  4    -0.942 387.16 1792.57 321.354 245.086 535.648 1365.715 0.0033  
301 2008  5    -0.355 388.50 1796.43 321.420 244.914 535.399 1365.717 0.0031  
302 2008  6     0.128 387.88 1791.80 321.447 244.676 535.128 1365.673 0.0031  
303 2008  7     0.003 386.42 1782.93 321.372 244.434 535.026 1365.672 0.0033  
304 2008  8    -0.266 384.15 1779.88 321.405 244.200 535.072 1365.657 0.0036  
305 2008  9    -0.643 383.09 1795.08 321.529 244.083 535.048 1365.665 0.0043  
306 2008 10    -0.780 382.99 1814.18 321.796 244.080 534.927 1365.676 0.0046  
307 2008 11    -0.621 384.13 1812.37 322.013 244.225 534.906 1365.707 0.0048  
308 2008 12    -0.666 385.56 1812.88 322.182 244.204 535.005 1365.693 0.0046  
    Temp  Predict  
285 0.601 0.4677808
286 0.498 0.4435404
287 0.435 0.4265541
288 0.466 0.4299162
289 0.372 0.4455113
290 0.382 0.4151422
291 0.394 0.4097367
292 0.358 0.3839390
293 0.402 0.3255595
294 0.362 0.3274147
295 0.266 0.3231401
296 0.226 0.3316704
297 0.074 0.3522134
298 0.198 0.3313129
299 0.447 0.3142112
300 0.278 0.3703410
301 0.283 0.4162213
302 0.315 0.4391458
303 0.406 0.4237965
304 0.407 0.3913679
305 0.378 0.3587615
306 0.440 0.3451991
307 0.394 0.3607087
308 0.330 0.3638076

In [22]:
SSE = sum((Test$Predict - Test$Temp)^2)
baseline = mean(Train$Temp, na.rm=T)
SST = sum((Test$Temp - baseline)^2)
R.squared = 1-SSE/SST
R.squared

[1] 0.6286051